source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/form/DispositionSaleForm.tsx

```diff
@@ -0,0 +1,214 @@
+import { getIn, useFormikContext } from 'formik';
+
+import { FastCurrencyInput, FastDatePicker } from '@/components/common/form';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import { FastDateYearPicker } from '@/components/common/form/FastDateYearPicker';
+import { PrimaryContactSelector } from '@/components/common/form/PrimaryContactSelector/PrimaryContactSelector';
+import { YesNoSelect } from '@/components/common/form/YesNoSelect';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { DispositionSaleFormModel } from '@/features/mapSideBar/disposition/models/DispositionSaleFormModel';
+import { useModalContext } from '@/hooks/useModalContext';
+import { SystemConstants, useSystemConstants } from '@/store/slices/systemConstants';
+import { getCurrencyCleanValue, stringToBoolean } from '@/utils/formUtils';
+
+import { useCalculateNetProceeds } from '../hooks/useCalculateNetProceedsBeforeSPP';
+import DispositionSalePurchaserSubForm from './DispositionSalePurchasersSubForm';
+
+export interface IDispositionSaleFormProps {
+  dispostionSaleId: number | null;
+}
+
+const DispositionSaleForm: React.FunctionComponent<
+  React.PropsWithChildren<IDispositionSaleFormProps>
+> = ({ dispostionSaleId }) => {
+  const formikProps = useFormikContext<DispositionSaleFormModel>();
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const isGstRequired = getIn(formikProps.values, 'isGstRequired');
+  const dispositionPurchaserAgent = getIn(formikProps.values, 'dispositionPurchaserAgent');
+  const dispositionPurchaserSolicitor = getIn(formikProps.values, 'dispositionPurchaserSolicitor');
+
+  const { getSystemConstant } = useSystemConstants();
+  const gstConstant = getSystemConstant(SystemConstants.GST);
+  const gstDecimal = gstConstant !== undefined ? parseFloat(gstConstant.value) / 100 : undefined;
+
+  useCalculateNetProceeds(isGstRequired);
+
+  // Functions
+  const onFinalSaleAmountUpdated = (newValue: string): void => {
+    const isGstRequired = getIn(formikProps.values, 'isGstRequired');
+    const cleanValue = getCurrencyCleanValue(newValue);
+
+    setGSTDerivedAmountFields(isGstRequired, cleanValue);
+  };
+
+  const onUpdateGstApplicable = (gstOption: string): void => {
+    const isGstRequired = stringToBoolean(gstOption);
+    const taxCollectedAmountValue = getIn(formikProps.values, 'gstCollectedAmount');
+
+    if (!isGstRequired) {
+      setModalContent({
```

Comment:
this looks like it isn't covered in tests, but is one of the more complex workflows in this logic, probably should be covered.

Reply:
updated